org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLFoldings.java

```diff
@@ -75,6 +75,7 @@ public List<FoldingRange> getFoldingRanges(TextDocument document, XMLFoldingSett
 			while (token != TokenType.EOS) {
 				cancelChecker.checkCanceled();
 				switch (token) {
+				case DTDStartDoctypeTag:
 				case StartTag: {
 					String tagName = scanner.getTokenText();

```

Comment:
Do you need tag name for the case of DOCTYPE?

Reply:
I believe so, its used to add the start tag to the stack with `stack.add(new TagInfo(startLine, tagName));` so we can keep track of the startline. I could also set a default value for doctype when creating a new TagInfo instead of using `scanner.getTokenText();`, but I'm not sure if that's necessary.

Reply:
> I could also set a default value for doctype when creating a new TagInfo instead of using scanner.getTokenText();, but I'm not sure if that's necessary.

I agree but it is just for the understand. It is strange to have tagName for DOCTYPE. But we can let like this.